registry/config_file_test.go

```diff
@@ -0,0 +1,135 @@
+package registry
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"testing"
+
+	"github.com/docker/docker/pkg/homedir"
+)
+
+func TestMissingFile(t *testing.T) {
+	tmpHome, _ := ioutil.TempDir("", "config-test")
+
+	config, err := LoadConfig(tmpHome)
+	if err != nil {
+		t.Fatalf("Failed loading on missing file: %q", err)
+	}
+
+	// Now save it and make sure it shows up in new form
+	err = config.Save()
+	if err != nil {
+		t.Fatalf("Failed to save: %q", err)
+	}
+
+	buf, err := ioutil.ReadFile(filepath.Join(tmpHome, CONFIGFILE))
+	if !strings.Contains(string(buf), `"auths":`) {
+		t.Fatalf("Should have save in new form: %s", string(buf))
+	}
+}
+
+func TestEmptyFile(t *testing.T) {
+	tmpHome, _ := ioutil.TempDir("", "config-test")
+	fn := filepath.Join(tmpHome, CONFIGFILE)
+	ioutil.WriteFile(fn, []byte(""), 0600)
+
+	_, err := LoadConfig(tmpHome)
+	if err == nil {
+		t.Fatalf("Was supposed to fail")
+	}
+}
+
+func TestEmptyJson(t *testing.T) {
+	tmpHome, _ := ioutil.TempDir("", "config-test")
+	fn := filepath.Join(tmpHome, CONFIGFILE)
+	ioutil.WriteFile(fn, []byte("{}"), 0600)
+
+	config, err := LoadConfig(tmpHome)
+	if err != nil {
+		t.Fatalf("Failed loading on empty json file: %q", err)
+	}
+
+	// Now save it and make sure it shows up in new form
+	err = config.Save()
+	if err != nil {
+		t.Fatalf("Failed to save: %q", err)
+	}
+
+	buf, err := ioutil.ReadFile(filepath.Join(tmpHome, CONFIGFILE))
+	if !strings.Contains(string(buf), `"auths":`) {
+		t.Fatalf("Should have save in new form: %s", string(buf))
+	}
+}
+
+func TestOldJson(t *testing.T) {
+	if runtime.GOOS == "windows" {
+		return
+	}
+
+	tmpHome, _ := ioutil.TempDir("", "config-test")
+	defer os.RemoveAll(tmpHome)
+
+	homeKey := homedir.Key()
+	homeVal := homedir.Get()
+
+	defer func() { os.Setenv(homeKey, homeVal) }()
+	os.Setenv(homeKey, tmpHome)
+
+	fn := filepath.Join(tmpHome, OLD_CONFIGFILE)
+	js := `{"https://index.docker.io/v1/":{"auth":"am9lam9lOmhlbGxv","email":"@example.com"}}`
```

Comment:
Not sure how I managed to do that but I fixed it. thanks!

> On Apr 20, 2015, at 3:56 PM, Sebastiaan van Stijn <notifications@github.com> wrote:
> 
> In registry/config_file_test.go <https://github.com/docker/docker/pull/12009#discussion_r28722782>:
> 
> > +func TestOldJson(t *testing.T) {
> > +	if runtime.GOOS == "windows" {
> > +		return
> > +	}
> > +
> > +	tmpHome, _ := ioutil.TempDir("", "config-test")
> > +	defer os.RemoveAll(tmpHome)
> > +
> > +	homeKey := homedir.Key()
> > +	homeVal := homedir.Get()
> > +
> > +	defer func() { os.Setenv(homeKey, homeVal) }()
> > +	os.Setenv(homeKey, tmpHome)
> > +
> > +	fn := filepath.Join(tmpHome, OLD_CONFIGFILE)
> > +	js := `{"https://index.docker.io/v1/":{"auth":"am9lam9lOmhlbGxv","email":"@example.com"}}`
> @duglin <https://github.com/duglin> Looks like you created an incomplete email address after changing it from @gmail.com
> 
> Not sure it's a problem here, just noting it, just in case :)
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/docker/docker/pull/12009/files#r28722782>.
>